### PR TITLE
adding timechart tests

### PR DESCRIPTION
**Pull request by rlgomes**
more missing app tests for outrigger.

### CODE REVIEW

**Comment by rlgomes** [comment]
@mnibecker @go-oleg @mstemm

File: test/app/timechart.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+let OutriggerTester = require('./lib/outrigger-tester');
+let path = require('path');
+let expect = require('chai').expect;
+
+const TEST_TIMEOUT = 10000;
+
+describe('barchart', function() {
```
Comment (mstemm) [inline test/app/timechart.spec.js]: Shouldn't this be 'timechart' ?

**Comment by mstemm** [comment]
Seems like a good start. Maybe you could add a testcase for events overlaid on a timechart as well?

**Comment by rlgomes** [comment]
@mstemm it is true that the events do not have any other way of being visualized if not on an exisitng timechart so it makes sense to do it here and I'll fix that barchart/timechart typo.

**Comment by mnibecker** [comment]
I'm not a fan of getting this deep with system tests, especially in outrigger. Really worried about repeating the testing of Jut 1.0, not to mention that this testing should probably be done in `juttle/juttle-viz` instead

**Comment by rlgomes** [comment]
@mnibecker without these tests we'd have (and have) broken integrations between juttle + juttle-viz + juttle-client-library and I'm not even in the realm yet of figuring out what happens when you throw any of the adapters in the mix. I'm not going to force real browser tests down the pipe to juttle-client-library, juttle-viz as that would make development there slower with no real benefit since the real important thing is that the apps built with these various components work as expected. So the idea here is to build simple and targeted "app tests" that verify our apps built with these components (currently only outrigger) can produce a valid result for the more common use cases.

If you have a better approach please share it as right now we've had various bugs found that could definitely deter others from using the product and this is the only approach that I see as a feasible solution.

**Comment by mnibecker** [comment]
Instead we could write an integration test that would run the programs you specified and verify that we're getting the correct points in the correct order. With that part covered, these tests really aren't testing juttle-client-library, they're testing juttle-viz and making sure they're rending correctly. This repo should trust that juttle-viz is working correctly.

**Comment by rlgomes** [comment]
This is totally not just testing juttle-viz as its verifying that when you run a juttle program in outrigger that outrigger runs that against the current juttle runtime and is able to produce data + sink information that is then interpreted by the juttle-client-library and in turn the juttle-client-library has to connect that data along with the sink info (options, etc) to the correct visualization in juttle-viz which then renders something on a HTML page that has to be setup correctly (by juttle-client-library) to present the data to the end user who then can interact with said visualization. 

With all of that said there are so many handoffs between all of these components that there is no way you'll convince me that if each component is tested perfectly the integrated product built using those components would be bug free. This is also confirmed by the various bugs found in the past few weeks that highlight this exact problem.

**Comment by mnibecker** [comment]
"Various Bugs in the past few weeks" - What were the bugs?

A lot of the steps you're are talking about can be tested in isolation without having to bring in system tests that: 1) Take a long time to run and 2) Are super fragile

Can we talk about this in person before we go ahead with this strategy?

**Comment by rlgomes** [comment]
@mnibecker set something up for Monday in the office with the interested parties.

**Comment by mnibecker** [comment]
sounds good

**Comment by dmehra** [comment]
We can all talk next week, but i want to address the "super fragile" concern here for the record. The takeaway from Jut 1.0 testing experience was that 
(a) the system test approach gave us very good test coverage and ensured we never put out broken builds; 
(b) testing *live* charts and interactive features in end-to-end selenium way requires very special care, and can otherwise be fragile; 
(c) having a solid suite of unit tests for the visualizations would remove the need for system tests to cover all cases of chart parameters, etc. 

I think @rlgomes is choosing a good strategy with having very basic system tests that verify that the combo of juttle+outrigger+viz is working correctly. @mnibecker is right that we should trust juttle-viz to have test coverage for all the different parameters, but there needs to be a setup where we check the end-to-end operation. 

That should cover error paths too - not "does outrigger report each and every possible error" but "does outrigger surface a juttle runtime error to the user? a view error? a syntax error?" 3 cases, done.

For interactivity such as tooltips and zooming, we may leave that not covered by automated tests at all for now, it's a resource issue.

**Comment by rlgomes** [comment]
to be revived at some point.

**Pull request closed without merging**